engine/classes/Elgg/Notifications/Event.php

```diff
@@ -110,6 +119,23 @@ public function getAction() {
 	public function getDescription() {
 		return "{$this->action}:{$this->object_type}:{$this->object_subtype}";
 	}
+
+	/**
+	 * Export
+	 * @return \stdClass
+	 */
+	public function toObject() {
+		$obj = new \stdClass();
+		$vars = get_object_vars($this);
+		foreach ($vars as $key => $value) {
+			if (is_callable([$value, 'toObject'])) {
```

Comment:
Make sure $value is an object first. If the value happens to match a classname this could cause trouble.